hooks/checkout

```diff
@@ -77,24 +81,44 @@ setup_git_repo() {
   set -e
 }
 
-i=0
-echo "--- :git: Checking out repository"
-while true; do
-  url="$(get_indirected_env "BUILDKITE_PLUGIN_SMOOTH_CHECKOUT_CONFIG_${i}_URL")"
-  ref="$(get_indirected_env "BUILDKITE_PLUGIN_SMOOTH_CHECKOUT_CONFIG_${i}_REF")"
-  if [ -z "${url}" ]; then
+CHECKED_OUT_REPO_NAME=""
+checkout_repo() {
+  local config_prefix="$1"
+  local index=0
+
+  echo "--- :git: Checking out repository"
+  while true; do
+    url="$(get_indirected_env "${config_prefix}_${index}_URL")"
+    ref="$(get_indirected_env "${config_prefix}_${index}_REF")"
+    if [ -z "${url}" ]; then
+        exit 1
+    fi
+
+    setup_git_repo "$url" "$ref"
+    EXIT_STATUS=$?
     if [[ $EXIT_STATUS -eq 0 ]]; then
+      CHECKED_OUT_REPO_NAME="$(get_repo_name "$url")"
       break
-    else
-      exit $EXIT_STATUS
     fi
-  fi
 
-  setup_git_repo "$url" "$ref"
-  EXIT_STATUS=$?
-  if [[ $EXIT_STATUS -eq 0 ]]; then
+    index=$((index+=1))
+  done
+}
+
+PLUGIN_PREFIX="BUILDKITE_PLUGIN_SMOOTH_CHECKOUT"
+
+index=0
+while true; do
+  config_prefix="${PLUGIN_PREFIX}_REPOS_${index}_CONFIG"
+  if [[ -z "$(get_indirected_env "${config_prefix}_0_URL")" || "$index" -gt 5 ]]; then
```

Comment:
Why this restriction - `"$index" -gt 5`?

Reply:
Oh I forgot to remove this, it was a fallback during testing so that the job doesn't get stuck in an infinite loop